frontend/src/components/SearchImage/SearchImage.jsx

```diff
@@ -1,28 +1,49 @@
-import React, { useRef } from 'react'
-import './SearchImage.scss';
+import React, { useRef, useState, useEffect } from "react";
+import "./SearchImage.scss";
 
 const SearchImage = ({ onChange, setInputValue, value }) => {
-    const searchRef = useRef();
+  const searchRef = useRef();
+  const [isSearchBarFull, setSearchBarFull] = useState(true);
 
-    function handleSearchSubmit(e) {
-        e.preventDefault();
-        const searchTerm = searchRef.current.value;
-        onChange(searchTerm);
+  function handleSearchSubmit(e) {
+    e.preventDefault();
+    const searchTerm = searchRef.current.value;
+    onChange(searchTerm);
+  }
+
+  const isSticky = () => {
+    const scrollTop = window.scrollY;
+    if (scrollTop > 20) {
+      setSearchBarFull(false);
+    } else {
+      setSearchBarFull(true);
     }
+  };
+  useEffect(() => {
+    window.addEventListener("scroll", () => isSticky());
+
+    return () => {
+      window.removeEventListener("scroll", () => isSticky());
+    };
+  });
 
-    return (
-        <div className="search_input-wrapper">
-            <form onSubmit={handleSearchSubmit}>
-                <input
-                    ref={searchRef}
-                    value={value}
-                    onChange={e => setInputValue(e.target.value)}
-                    className="search_input"
-                    type="text"
-                    placeholder="Search..." />
-            </form>
-        </div>
-    )
-}
+  return (
+    <div className="search_input-wrapper">
+      <form
+        onSubmit={handleSearchSubmit}
+        className={isSearchBarFull ? `form-style-one` : `form-style-second`}
```

Comment:
Can you rename classnames as 'search-bar-lg' and 'search-bar-md'? Or something similar

Reply:
resolve

Reply:
did some improvements here